pkg/broker/broker.go

```diff
@@ -76,23 +82,42 @@ func (br *Broker) ListenForClients() {
 		case sb := <-br.NewClients:
 			br.clients[sb] = true
 			cc, ok := br.clientsCount.Load(sb.ID)
-			if ok {
+			if ok && cc.(int) > 0 {
 				br.clientsCount.Store(sb.ID, cc.(int)+1)
 			} else {
 				br.clientsCount.Store(sb.ID, 1)
+				br.CloseClientC[sb.ID] = make(chan struct{})
+				tl, err := tailer.New(sb.Extra)
+				go func() {
+					br.Log.Printf("[tailer] Starting for %s", sb.ID)
+
+					scanner := bufio.NewScanner(tl)
+					for scanner.Scan() {
+						br.Notifier <- &Event{Msg: []byte(scanner.Text()), ID: sb.ID}
+					}
+				}()
+				go func() {
+					<-br.CloseClientC[sb.ID]
+					br.Log.Printf("[tailer] Exiting for: %s", sb.ID)
+					err = tl.Close()
+					if err != nil {
+						br.Log.Print(err)
+					}
+				}()
 			}
 			br.Log.Printf("[broker] Client added. %d registered clients", len(br.clients))
 		case sb := <-br.ClosingClients:
```

Comment:
Rename this variable to "client".

Reply:
done, it was about time we do this :smile: